h5netcdf/core.py

```diff
@@ -674,6 +690,11 @@ def resize_dimension(self, dimension, size):
         # Resize the dimension.
         self._current_dim_sizes[dimension] = size
 
+        # If not a variable but dimension -> resize
+        if dimension not in self.variables and dimension in self._h5group:
+            self._h5group[dimension].resize((size,))
```

Comment:
This looks sensible, but is it related somehow to the other changes? It might warrant another unit-test.

Reply:
If only a dimension is created, the corresponding Dataset will be created at the same time. Before it would only be created on `flush`.

Now, when we create a dimension and resize it afterwards, we have to resize here since the Dataset already exists. 

I'll take a look and come up with some more tests with this. Good chance that this will lead to some more refactoring.

Reply:
@shoyer I've added tests, which checks correct resizing of dimensions vs. variables.